### PR TITLE
Document rebase script

### PR DESCRIPTION
**Pull request by fzdarsky**
This is the documentation of the rebase process at the state after #737 merges.

Closes [USHIFT-65](https://issues.redhat.com//browse/USHIFT-65)

### CODE REVIEW

File: docs/rebase.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+git apply scripts/rebase_patches/*
+script/rebase.sh generated-apis
+git add vendor
+git commit -m "update vendoring"
+```
+
+These patches are produced by identifying the missing patches [TODO: describe process] and transforming those patches to apply against `/vendor` instead of o/k's staging dir [TODO: describe process]. This process requires some manual work but should be necessary rarely (likely after OpenShift rebases onto a upstream version).
+
+## Rebasing `Makefile`, `Dockerfiles`, and `.spec` file
+When updating to a new minor version of OpenShift, update the `Makefile` [here](https://github.com/openshift/microshift/blob/main/Makefile#L16) and [here](https://github.com/openshift/microshift/blob/main/Makefile#L55-L66) and also check whether the Dockerfiles need a new base image (e.g. [this](https://github.com/openshift/microshift/blob/main/packaging/images/openshift-ci/Dockerfile)) and the RPM `.spec` file needs updates (e.g. [this](https://github.com/openshift/microshift/blob/main/packaging/rpm/microshift.spec))
```
Comment (ggiguash) [inline docs/rebase.md:73]: The lines in Makefile are likely to go out of sync quickly. Let's use a description of those places?

File: docs/rebase.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+### Component images
+To update the component image references of the MicroShift release, run:
+
+```
+./scripts/rebase.sh images
+git add pkg/release
+git commit -m "update component images"
+```
+
+Note that as long as we don't use 100% OpenShift-built images, you may need to bump the `pause` and network and storage provider images manually.
```
Comment (ggiguash) [inline docs/rebase.md:92]: It's not clear (to me at least) what such a "bump" would require

File: docs/rebase.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+Embed the assets into the binary and commit the result:
+
+```
+./scripts/bindata.sh
+git add assets pkg/assets
+git commit -m "update manifests"
+```
+
+### Component configs
+The last step isn't automated at all yet, which is to compare whether the config parameters of embedded component changed.
```
Comment (ggiguash) [inline docs/rebase.md:125]: Can we add at least an example of what needs to be compared to what?

**Review by ggiguash — CHANGES_REQUESTED** [review]
Overall, looking great! Some minor comments, mostly for adding a bit more information.

File: docs/rebase.md (hunk 1 above)
Comment (fzdarsky) [reply docs/rebase.md]: Good point. Fixed this.

File: docs/rebase.md (hunk 2 above)
Comment (fzdarsky) [reply docs/rebase.md]: Removed that note as our change to OpenShift images will make this irrelevant anyway.

File: docs/rebase.md (hunk 3 above)
Comment (fzdarsky) [reply docs/rebase.md]: Added this.

**Review by ggiguash — APPROVED** [review]

**Review by sallyom — COMMENTED** [review]
This has all the necessary details and will make rebases a much kinder & simpler ordeal. I'm going through the steps now to ensure everything is working as intended.

**Comment by sallyom** [comment]
Are we oppposed to adding a section (or separate .md file) for building a community version, something like:


# Build MicroShift From a Community OKD Release

The rebase script can also be used to build a community version of MicroShift that does not require access to internal image registries. To build the community version, rebase against an [OKD image found here](https://origin-release.ci.openshift.org/#4-stable) by following the same rebase script, and downloading the OKD release image [like so](#downloading-the-target-openshift-release) instead of an OpenShift release. This build is not supported or maintained and there is no guarantee that OKD releases will continue to mirror OpenShift releases.

**Comment by fzdarsky** [comment]
> Are we oppposed to adding a section (or separate .md file) for building a community version, something like:

That's a good idea, let me include that.

File: docs/rebase.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
+# Rebasing MicroShift
+## Overview
+
+MicroShift repackages a minimal core of OpenShift: A given MicroShift release is built from the same content as the corresponding OpenShift release, adding a small amount of "glue logic".
+
+Therefore, on a high level a rebase of MicroShift onto a newer OpenShift release starts from an given OpenShift release image and involves the following steps:
```
Comment (dhellmann) [inline docs/rebase.md:6]:
```suggestion
Therefore, on a high level a rebase of MicroShift onto a newer OpenShift release starts from a given OpenShift release image and involves the following steps:
```

**Review by dhellmann — APPROVED** [review]
Very clear instructions. I found one typo, but that can be fixed in another PR if there's no other reason to update this one.

File: docs/rebase.md (hunk 4 above)
Comment (fzdarsky) [reply docs/rebase.md]: Thank you, included that in the update.

**Review by sallyom — APPROVED** [review]
https://github.com/openshift/microshift/pull/737 is approved & tagged, tagging this as well! 

/lgtm

**Pull request merged**